File: C# Net/Project/FilterData.cs
using System.Linq;

namespace Testing.ValuesGenerator
{
    public class FilterData
    {
        public string[] WhiteList;
        public string[] BlackList;
        public string[] AvailableGroups;
        public string Breadcrumb = "";

        public FilterData() { }

        public FilterData(string[] whiteList, string[] blackList, string[] availableGroups, string breadcrumb)
        {
            WhiteList = whiteList;
            BlackList = blackList;
            AvailableGroups = availableGroups;
            Breadcrumb = breadcrumb;
        }

        public FilterData(FilterData filterData, string fieldName) : 
            this(filterData.WhiteList, filterData.BlackList, filterData.AvailableGroups, filterData.Breadcrumb + "." + fieldName) { }

        public string GetBreadCrumbForCheck()
        {
            return (Breadcrumb.Last() == '.')
                ? Breadcrumb
                : Breadcrumb + ".";
        }

        public string GetCheckValue(string value)
        {
            if (value.First() != '.')
                value = "." + value;
            if (value.Last() != '.')
                value = value + ".";
            return value;
        }

        public bool AllowGeneration()
        {
            if (WhiteList != null && WhiteList.Any())
                return WhiteList.Any(val => GetBreadCrumbForCheck().Contains(GetCheckValue(val)));
            if (BlackList != null && BlackList.Any())
                return !BlackList.Any(val => GetBreadCrumbForCheck().Contains(GetCheckValue(val)));
            return true;
        }
    }
}

File: Java/src/main/java/generator/FilterData.java
package generator;

import funcInterfaces.*;

/**
 * Created by 12345 on 28.12.2014.
 */
public class FilterData {
    public String[] whiteList;
    public String[] blackList;
    public FuncTT<String, Boolean> generationRule;
    public String[] availableGroups;
    public String breadcrumb = "*";

    public FilterData() { }

    public FilterData(String[] whiteList, String[] blackList, FuncTT<String, Boolean> generationRule,
                      String[] availableGroups, String breadcrumb)
    {
        this.whiteList = whiteList;
        this.blackList = blackList;
        this.generationRule = generationRule;
        this.availableGroups = availableGroups;
        this.breadcrumb = breadcrumb;
    }

    public FilterData(FilterData filterData, String fieldName) {
        this(filterData.whiteList, filterData.blackList, filterData.generationRule, filterData.availableGroups, filterData.breadcrumb + "." + fieldName);
    }

    public boolean allowGeneration() throws Exception {
        if (generationRule != null)
            return generationRule.invoke(breadcrumb);
        if (whiteList != null && whiteList.length > 0) {
            for(String value : whiteList)
                if (breadcrumb.contains(value))
                    return true;
            return false;
        }
        if (blackList != null && blackList.length > 0) {
            for(String value : blackList)
                if (breadcrumb.contains(value))
                    return false;
            return true;
        }
        return true;
    }

}
